server/action_creators.js

```diff
@@ -53,40 +53,23 @@ export function addOrder(entries) {
 }
 
 export function changeEntryStatus(orderId, entryIndex, status) {
-  return (dispatch, getState) => {
-    const orders = getState().orders;
-    const orderIndex = orders.findIndex((order) => order.id === orderId);
-
-    return Order.updateEntry(orderId, entryIndex, { status })
-      .then((response) => {
-        const updatedOrders = [
-          ...orders.slice(0, orderIndex),
-          response[0],
-          ...orders.slice(orderIndex + 1)
-        ];
-
-        dispatch(setState({ orders: updatedOrders }));
-      });
-  };
+  return transactAndDispatch(orderId, Order.updateEntry(orderId, entryIndex, { status }));
 }
 
 export function addEntriesToOrder(orderId, newEntries) {
-  return (dispatch, getState) => {
-    return Order.addEntries(orderId, newEntries).then(() => {
-      return Order.getOrders();
-    }).then((orders) => {
-      dispatch(setState({ orders }));
-    });
-  };
+  return transactAndDispatch(orderId, Order.addEntries(orderId, newEntries));
 }
 
 export function setReadyForBill(orderId) {
+  return transactAndDispatch(orderId, Order.updateStatus(orderId, constants.READY_FOR_BILL));
+}
+
+function transactAndDispatch(orderId, transaction) {
```

Comment:
Yeah. We can definitely abstract some more but I was thinking this could be a good start since we operate a lot on orders and would reduce some of the obvious repetition.